src/utilities/new-sequence-editor/time-utils.ts

```diff
@@ -1,137 +1,212 @@
-export const ABSOLUTE_TIME = /^(\d{4})-(\d{3})T(\d{2}):(\d{2}):(\d{2})(?:\.(\d{3}))?$/g;
+import { CustomErrorCodes, type ErrorCode } from '../../workers/customCodes';
 
-export const RELATIVE_TIME = /([0-9]{3}T)?([0-9]{2}):([0-9]{2}):([0-9]{2})(\.[0-9]+)?$/g;
-export const RELATIVE_SIMPLE = /(\d+)(\.[0-9]+)?$/g;
+export enum TimeTypes {
+  ABSOLUTE = 'absolute',
+  EPOCH = 'epoch',
+  EPOCH_SIMPLE = 'epoch_simple',
+  RELATIVE = 'relative',
+  RELATIVE_SIMPLE = 'relative_simple',
+}
+
+export const ABSOLUTE_TIME = /^(\d{4})-(\d{3})T(\d{2}):(\d{2}):(\d{2})(?:\.(\d{3}))?$/;
 
-export const EPOCH_TIME = /(^[+-]?)([0-9]{3}T)?([0-9]{2}):([0-9]{2}):([0-9]{2})(\.[0-9]+)?$/g;
-export const EPOCH_SIMPLE = /(^[+-]?)(\d+)(\.[0-9]+)?$/g;
+export const RELATIVE_TIME = /^([0-9]{3}T)?([0-9]{2}):([0-9]{2}):([0-9]{2})(\.[0-9]+)?$/;
```

Comment:
Is it necessary for other places to be using these regexs? Could the `testTime` function be repurposed to instead of taking the regex as a parameter, take the `TimeType` as a parameter to be used in combination with the new `getTimeRegex` function?